### PR TITLE
move unnecessary dict trainer_options

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?
move unnecessary dict trainer_options in test which causes problem with refactoring

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: tests/base/utils.py [hunk 1]
```diff
@@ -59,8 +57,10 @@ def run_model_test(trainer_options, model, on_gpu=True):
     checkpoint = init_checkpoint_callback(logger)
 
     # add these to the trainer options
-    trainer_options['checkpoint_callback'] = checkpoint
-    trainer_options['logger'] = logger
+    trainer_options.update(dict(
```
Comment (justusschock) [inline tests/base/utils.py:60]: There is no need to wrap this in a dict, update can deal with kwargs

File: tests/loggers/test_wandb.py [hunk 2]
```diff
@@ -47,11 +47,9 @@ class Experiment:
 
     logger = WandbLogger(id='the_id', offline=True)
 
-    trainer_options = dict(max_epochs=1, logger=logger)
-
-    trainer = Trainer(**trainer_options)
+    trainer = Trainer(max_epochs=1, logger=logger)
     # Access the experiment to ensure it's created
-    trainer.logger.experiment
+    assert trainer.logger.experiment
```
Comment (justusschock) [inline tests/loggers/test_wandb.py:52]: Maybe add an assertion message here?

File: tests/models/test_restore.py [hunk 3]
```diff
@@ -236,11 +234,13 @@ def test_dp_resume(tmpdir):
 
     # init new trainer
     new_logger = tutils.get_default_testtube_logger(tmpdir, version=logger.version)
-    trainer_options['logger'] = new_logger
-    trainer_options['checkpoint_callback'] = ModelCheckpoint(tmpdir)
-    trainer_options['train_percent_check'] = 0.5
-    trainer_options['val_percent_check'] = 0.2
-    trainer_options['max_epochs'] = 1
+    trainer_options.update(dict(
```
Comment (justusschock) [inline tests/models/test_restore.py:237]: also no need to wrap this as a dict.

File: tests/trainer/test_dataloaders.py [hunk 4]
```diff
@@ -36,63 +42,13 @@ class CurrentTestModel(
...
-        max_epochs=1,
-        train_percent_check=-0.1,
-    )
-
-    # fit model
-    trainer = Trainer(**trainer_options)
-
-    with pytest.raises(ValueError):
-        trainer.fit(model)
-
```
Comment (awaelchli) [inline tests/trainer/test_dataloaders.py:53]: just double checking, was it intentional to remove these tests? was it duplicated?

Comment (Borda) [reply tests/trainer/test_dataloaders.py:53]: No test shall be removed unless it was merged with using pytest parameters...

Comment (Borda) [reply tests/trainer/test_dataloaders.py:53]: No test shall be removed unless it was merged with using pytest parameters...

Comment (awaelchli) [reply tests/trainer/test_dataloaders.py:53]: oh yes, missed that, it is parametrized, very nice!

**Pull request merged**